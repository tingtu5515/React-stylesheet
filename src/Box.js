/**
 * @flow
 */

import Element from './Element';
import stylesheet from './Stylesheet';

let boxStylesheet = stylesheet('Box', {
  base: {
    boxSizing: 'border-box',
    position: 'relative',

    overflow: 'hidden',

    margin: 0,
    padding: 0,

    display: 'flex',
    alignItems: 'stretch',
    flexBasis: 'auto',
    flexShrink: 0,

    minHeight: 0,
    minWidth: 0
  }
});

class Box extends Element {

  static className = boxStylesheet.toClassName();

  componentWillMount() {
    boxStylesheet.inject();
  }
}

export class VBox extends Box {

  static defaultProps = {
    flexDirection: 'column',
  };

}

export class HBox extends Box {

  static defaultProps = {
    flexDirection: 'row',
  };
}
