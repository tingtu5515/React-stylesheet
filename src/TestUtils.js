/**
 * @flow
 */

import {CSSClassRepr, CSSClassJoinRepr} from './CSSClassRepr';

export const snapshotSerializer = {
  // $ExpectError: ...
  print(val, serialize, _indent) {
    if (val instanceof CSSClassRepr) {
      const repr = serialize(val.repr);
      return repr.replace(/^Object {/, `${val.reprName} {`);
    } else if (val instanceof CSSClassJoinRepr) {
      const repr = serialize(val.styles);
      return repr.replace(/^Array \[/, 'StyleJoin [');
    } else {
      return serialize(val);
    }
  },

  // $ExpectError: ...
  test(val) {
    return val instanceof CSSClassRepr || val instanceof CSSClassJoinRepr;
  },
};
