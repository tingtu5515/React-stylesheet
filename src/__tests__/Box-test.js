/**
 * @flow
 */

import React from 'react';
import renderer from 'react-test-renderer';

import {VBox, HBox} from '../Box';
import * as TestUtils from '../TestUtils';

expect.addSnapshotSerializer(TestUtils.snapshotSerializer);

describe('<HBox />', function() {
  test('render', function() {
    const tree = renderer.create(<HBox>Hello</HBox>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('<BBox />', function() {
  test('render', function() {
    const tree = renderer.create(<VBox>Hello</VBox>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
