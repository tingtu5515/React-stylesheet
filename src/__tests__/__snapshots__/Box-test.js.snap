// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BBox /> render 1`] = `
<div
  className={
    StyleJoin [
      CompiledCSSWithVariants {
        "base": Object {
          "alignItems": "stretch",
          "display": "flex",
          "flexBasis": "auto",
          "flexShrink": 0,
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
        },
      },
      PrecompiledCSS {
        "boxSizing": "border-box",
      },
      PrecompiledCSS {
        "flexDirection": "column",
      },
    ]
  }
  style={Object {}}
>
  Hello
</div>
`;

exports[`<HBox /> render 1`] = `
<div
  className={
    StyleJoin [
      CompiledCSSWithVariants {
        "base": Object {
          "alignItems": "stretch",
          "display": "flex",
          "flexBasis": "auto",
          "flexShrink": 0,
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
        },
      },
      PrecompiledCSS {
        "boxSizing": "border-box",
      },
      PrecompiledCSS {
        "flexDirection": "row",
      },
    ]
  }
  style={Object {}}
>
  Hello
</div>
`;
